adapter/internal/synchronizer/apis_fetcher.go

```diff
@@ -234,8 +234,17 @@ func FetchAPIsFromControlPlane(updatedAPIID string, updatedEnvs []string, envToD
 			break
 		} else if data.ErrorCode >= 400 && data.ErrorCode < 500 {
 			logger.LoggerSync.Errorf("Error occurred when retrieving API %q from control plane: %v", updatedAPIID, data.Err)
-			health.SetControlPlaneRestAPIStatus(false)
+
+			// If API is not found (404), then there is no point in setting the control plane status as unhealthy.
+			if data.ErrorCode != 404 {
+				health.SetControlPlaneRestAPIStatus(false)
```

Comment:
Do we set this back to healthy, in somewhere, if this has happened and resolved?

Reply:
Idea here is if the status code 401, 403 , we don't retry rather we make the adapter to be killed as it is unrecoverable. That was the initial thought. But now um wondering if we do such thing in first place. May be a log alert is the correct way to move forward.